.github/workflows/quality-assurance.yml

```diff
@@ -0,0 +1,86 @@
+name: quality-assurance
+
+on:
+  pull_request:
+
+env:
+  python-version: "3.9"
+  rust-version: "1.61.0"
+  node-version: "16"
+
+jobs:
+  check-quality-assurance:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+      - uses: actions/setup-python@d09bd5e6005b175076f227b13d9730d56e9dcfcb
+        with:
+          python-version: ${{ env.python-version }}
+
+      - uses: actions/setup-node@eeb10cff27034e7acf239c5d29f62154018672fd
+        with:
+          node-version: ${{ env.node-version }}
+
+      - name: Cache Rust
+        id: cache-rust
+        env:
+          cache-version: 1
+        uses: actions/cache@c3f1317a9e7b1ef106c153ac8c0f00fed3ddbc0d
+        with:
+          key: ${{ runner.os }}-rust-${{ env.cache-version }}.${{ env.rust-version }}-${{ hashFiles('Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-rust-${{ env.cache-version }}.${{ env.rust-version }}
+            ${{ runner.os }}-rust-${{ env.cache-version }}
+            ${{ runner.os }}-rust-
+          path: |
+            target
+            ~/.cargo/bin
+            ~/.cargo/registry/index
+            ~/.cargo/registry/cache
+            ~/.cargo/git/db
+
+      - if: ${{ steps.cache-rust.outputs.cache-hit == 'false' }}
+        uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af
+        with:
+          toolchain: ${{ env.rust-version }}
+          components: rustfmt, clippy
+          default: true
+
+      - name: Debug installed tools
+        run: |
+          python --version
+          rustc --version
+          cargo --version
+          node --version
+
+      - name: Debug rust cache
+        run: |
+          ls -la ~/.cargo/{.,bin,git,registry} target || true
+
+      - name: Check Commit Signature
+        run: python .github/scripts/check_commit_signature.py
+
+      - name: Check Newsfragment
+        if: |
+          ${{
+            startsWith(github.ref, 'refs/pull/')
+            && !(startsWith(github.ref_name, 'dependabot')
+              || startsWith(github.ref_name, 'yolo')
+              || startsWith(github.ref_name, 'release'))
+          }}
+        run: |
+          whereis git
+          git fetch origin master
+          python .github/scripts/check_news_fragment.py ${{ github.ref_name }}
+
+      - name: Run cargo build to prepare rust's cache
+        run: cargo build --workspace
+
+      - name: Patch pre-commit for line-ending
+        run: |
+          sed -i '/id: mixed-line-ending/a\      args:\n\        - "--fix=lf"' .pre-commit-config.yaml
+
+      - uses: pre-commit/action@646c83fcd040023954eafda54b4db0192ce70507
```

Comment:
This action depends itself of `actions/cache@v3` (see https://github.com/pre-commit/action/blob/main/action.yml).

Unfortunately we cannot pin the transitive dependency :(

Do you have an idea how action triggering sub action like this is common ? I would guess it's something rather rare (I've had a look at cache, setup-python and setup-node, which all don't do that).

Regarding what to do here, `actions/cache` is an official action from Github that is massively used. So the risk of it being used to leak data is very low (compared to using a action created by $RANDOM_DEV).

So I guess this is ok to use pre-commit/action here, but we should be careful of this and read the content of each action we want to add to our CI

Reply:
> We could also fork each actions used in our orga.

This seems like a good approach whenever we want to use exotic (i.e. not Github official) actions 👍